lib/elixir/lib/integer.ex

```diff
@@ -59,21 +61,29 @@ defmodule Integer do
 
   """
   @spec digits(non_neg_integer, pos_integer) :: [non_neg_integer, ...]
-  def digits(n, base \\ 10) when is_integer(n)    and n >= 0
-                            and  is_integer(base) and base >= 2 do
-    do_digits(n, base, [])
+  def digits(integer, base \\ 10)
+  def digits(0, base) when is_integer(base) and base >= 2,
```

Comment:
Great! Also I think we can use acc in these clauses, e.g. [1, 0 | acc]. It will optimize some extra cases.

> On 07 May 2016, at 18:47, eksperimental <notifications@github.com> wrote:
> 
> In lib/elixir/lib/integer.ex:
> 
> > @@ -59,21 +61,29 @@ defmodule Integer do
> >  
> >    """
> >    @spec digits(non_neg_integer, pos_integer) :: [non_neg_integer, ...]
> > -  def digits(n, base \\ 10) when is_integer(n)    and n >= 0
> > -                            and  is_integer(base) and base >= 2 do
> > -    do_digits(n, base, [])
> > +  def digits(integer, base \\ 10)
> > +  def digits(0, base) when is_integer(base) and base >= 2,
> thank you @lexmag for making me see this,
> have a look at how it looks now. WHAT A DIFFERENCE!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>